contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditBooleanMapModal.vue

```diff
@@ -0,0 +1,425 @@
+      },
+      validators: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data() {
+      return {
+        updateDescendants: false,
+        searchQuery: '',
+        error: '',
+        /**
+         * selectedValues is an object with the following structure:
+         * {
+         *  [optionId]: true | [nodeId1, nodeId2, ...]
+         * }
+         * If the value is true, it means that the option is selected for all nodes
+         * If the value is an array of nodeIds, it means that the option is selected
+         * just for those nodes
+         */
+        selectedValues: {},
+      };
+    },
+    computed: {
+      ...mapGetters('contentNode', ['getContentNodes']),
+      nodes() {
+        return this.getContentNodes(this.nodeIds);
+      },
+      isTopicSelected() {
+        return this.nodes.some(node => node.kind === ContentKindsNames.TOPIC);
+      },
+      filteredOptions() {
+        const searchQuery = this.searchQuery?.trim().toLowerCase();
+        if (!this.searchQuery) {
+          return this.options;
+        }
+        return this.options.filter(option => option.label.toLowerCase().includes(searchQuery));
+      },
+      flatList() {
+        if (!this.showHierarchy) {
+          return true;
+        }
+        return this.searchQuery && this.searchQuery.trim().length > 0;
+      },
+      dividerStyle() {
+        return {
+          border: 0,
+          borderBottom: `1px solid ${this.$themeTokens.fineLine}`,
+          margin: '1em 0',
+        };
+      },
+      autocompleteOptions() {
+        return [
+          {
+            value: MIXED,
+            label: this.$tr('mixedLabel'),
+            undeletable: true,
+          },
+          ...this.options,
+        ];
+      },
+      autocompleteValues() {
+        const selectedValues = Object.entries(this.selectedValues)
+          .filter(entry => entry[1] === true) // no mixed values
+          .map(([key]) => key);
+        if (Object.values(this.selectedValues).some(value => value !== true)) {
+          selectedValues.push(MIXED);
+        }
+        return selectedValues;
+      },
+    },
+    created() {
+      const optionsNodes = {};
+
+      this.nodes.forEach(node => {
+        Object.entries(node[this.field] || {})
+          .filter(entry => entry[1] === true)
+          .forEach(([optionId]) => {
+            optionsNodes[optionId] = optionsNodes[optionId] || [];
+            optionsNodes[optionId].push(node.id);
+          });
+      });
+
+      Object.entries(optionsNodes).forEach(([key, nodeIds]) => {
+        if (nodeIds.length === this.nodeIds.length) {
+          this.selectedValues[key] = true;
+        } else {
+          this.selectedValues[key] = nodeIds;
+        }
+      });
+    },
+    methods: {
+      ...mapActions('contentNode', ['updateContentNode', 'updateContentNodeDescendants']),
+      close() {
+        this.$emit('close');
+      },
+      isCheckboxSelected(optionId) {
```

Comment:
Not a blocker, but I think we could still refine the code structure here to be even much better and more clear(It seems to be a very important piece to the puzzle so it's important that its clear). With still lots of moving pieces in the bulk edit feature, we can revisit it later during the merge of the feature branch back to `unstable`